tests/testsuite/vendor.rs

```diff
@@ -0,0 +1,509 @@
+    );
+    p.cargo("vendor --respect-source-config").run();
+
+    let lock = p.read_file("vendor/bitflags/Cargo.toml");
+    assert!(lock.contains("version = \"0.8.0\""));
+}
+
+#[test]
+fn two_lockfiles() {
+    let p = project()
+        .no_manifest()
+        .file(
+            "foo/Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+
+                [dependencies]
+                bitflags = "=0.7.0"
+            "#,
+        )
+        .file("foo/src/lib.rs", "")
+        .file(
+            "bar/Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.1.0"
+
+                [dependencies]
+                bitflags = "=0.8.0"
+            "#,
+        )
+        .file("bar/src/lib.rs", "")
+        .build();
+
+    Package::new("bitflags", "0.7.0").publish();
+    Package::new("bitflags", "0.8.0").publish();
+
+    p.cargo("vendor --respect-source-config -s bar/Cargo.toml --manifest-path foo/Cargo.toml")
+        .run();
+
+    let lock = p.read_file("vendor/bitflags/Cargo.toml");
+    assert!(lock.contains("version = \"0.8.0\""));
+    let lock = p.read_file("vendor/bitflags-0.7.0/Cargo.toml");
+    assert!(lock.contains("version = \"0.7.0\""));
+
+    add_vendor_config(&p);
+    p.cargo("build").cwd("foo").run();
+    p.cargo("build").cwd("bar").run();
+}
+
+#[test]
+fn delete_old_crates() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+
+                [dependencies]
+                bitflags = "=0.7.0"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    Package::new("bitflags", "0.7.0").publish();
+    Package::new("log", "0.3.5").publish();
+
+    p.cargo("vendor --respect-source-config").run();
+    p.read_file("vendor/bitflags/Cargo.toml");
+
+    p.change_file(
+        "Cargo.toml",
+        r#"
+            [package]
+            name = "foo"
+            version = "0.1.0"
+
+            [dependencies]
+            log = "=0.3.5"
+        "#,
+    );
+
+    p.cargo("vendor --respect-source-config").run();
+    let lock = p.read_file("vendor/log/Cargo.toml");
+    assert!(lock.contains("version = \"0.3.5\""));
+    assert!(!p.root().join("vendor/bitflags/Cargo.toml").exists());
+}
+
+#[test]
+fn ignore_files() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+
+                [dependencies]
+                url = "1.4.1"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    Package::new("url", "1.4.1")
+        .file("src/lib.rs", "")
+        .file("foo.orig", "")
+        .file(".gitignore", "")
+        .file(".gitattributes", "")
+        .file(".cargo-ok", "")
```

Comment:
This line should be removed. `.crate` files do not contain `.cargo-ok`, that is added during extraction.  It fails on windows because windows has a rw lock, and won't let you overwrite it.

It took a while to diagnose this because TarError does not display its contained io::Error, thus hiding it (in this case, permission denied).  It would be nice if TarError either displayed it, or cooperated with failure's causal chain somehow.

Reply:
Heh excellent eyes as usual, I've avoided fixing the test and have hopefully updated the error printing code in Cargo to handle this better

Reply:
```
    Updating `C:\projects\cargo\target\cit\t1461\registry` index
 Downloading crates ...
  Downloaded url v1.4.1 (registry `C:\projects\cargo\target\cit\t1461\registry`)
error: failed to sync
Caused by:
  failed to download packages
Caused by:
  failed to download replaced source registry `https://github.com/rust-lang/crates.io-index`
Caused by:
  failed to unpack package `url v1.4.1 (registry `C:\projects\cargo\target\cit\t1461\registry`)`
Caused by:
  failed to unpack entry at `url-1.4.1/.cargo-ok`
Caused by:
  failed to unpack `C:\projects\cargo\target\cit\t1461\home\.cargo\registry\src\-3c7adbf626655e5c\url-1.4.1\.cargo-ok`
Caused by:
  failed to unpack `url-1.4.1/.cargo-ok` into `C:\projects\cargo\target\cit\t1461\home\.cargo\registry\src\-3c7adbf626655e5c\url-1.4.1\.cargo-ok`
Caused by:
  Access is denied. (os error 5)
```

showed at https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/24179518#L1933

yay!